plasmapy/diagnostics/tests/test_langmuir.py

```diff
@@ -1,40 +1,181 @@
+def characteristic_simulated():
+    r""""Create a simulated probe characteristic (provisional)"""
+
+    T_e_sim = 1 * u.eV
+    n_e_sim = 1e18 * u.m**-3
+    probe_area_sim = 1 * u.mm**2
+    I_e_sim = (n_e_sim * probe_area_sim * const.e *
+               np.sqrt(T_e_sim / (2 * np.pi * const.m_e)))
+
+    # Create bias array
+    bias_simarr = np.arange(-20, 15, 0.1) * u.V
+
+    # Calculate electron current and limit to electron saturation current
+    current_simarr = np.exp(const.e * bias_simarr / T_e_sim) * u.A
+    current_simarr[current_simarr > I_e_sim] = I_e_sim
+
+    # Add simulated linear sheath expansion current
+    current_simarr[current_simarr == I_e_sim] += (
+        bias_simarr[current_simarr == I_e_sim] * 0.0005 * u.A/u.V)
+
+    # Add simulated linear ion collection current
+    current_simarr[current_simarr < I_e_sim] += (
+        bias_simarr[current_simarr < I_e_sim] * 0.0001 * u.A/u.V)
+
+    return Characteristic(bias_simarr, current_simarr)
+
+
+@pytest.fixture
+def shuffle_characteristic(characteristic):
+    r""""Shuffle a given characteristic"""
+
+    _shuffle = sorted(np.arange(len(characteristic.bias)),
+                      key=lambda k: np.random.random())
+    U_shuffled = characteristic.bias[_shuffle]
+    I_shuffled = characteristic.current[_shuffle]
+    return Characteristic(U_shuffled, I_shuffled)
+
+
+class Test__swept_probe_analysis:
+    r"""Test the swept_probe_analysis function in langmuir.py"""
+
+    @staticmethod
+    def test_nan_area():
+        r"""Test error upon NaN area"""
+
+        with pytest.raises(ValueError):
+            swept_probe_analysis(characteristic, np.nan * u.m**2, 40 * u.u)
+
+    @staticmethod
+    def test_unit_conversion_error():
+        r"""Test error upon incorrect probe area unit"""
+
+        with pytest.raises(u.UnitConversionError):
+            swept_probe_analysis(characteristic, 1*u.m, 40 * u.u)
+
+    @staticmethod
+    def test_negative_area():
+        r"""Test error upon negative probe area"""
+
+        with pytest.raises(ValueError):
+            swept_probe_analysis(characteristic, -1 * u.m**2, 40 * u.u)
+
+    @staticmethod
+    def test_ion_mass_unit():
+        r"""Test equality of float and a.m.u. ion mass"""
+
+        sim = characteristic_simulated()
+
+        sim_result1 = swept_probe_analysis(
+            sim, 4*u.m**2, 40)
+
+        sim_result2 = swept_probe_analysis(
+            sim, 4*u.m**2, 40 * u.u)
+
+        errStr = (f"`swept_probe_analysis` should accept both floats and "
+                  f"a.m.u. Quantities as atomic gas mass input.")
+        for key in sim_result1:
+            assert (sim_result1[key] == sim_result2[key]).all(), errStr
+
+    @staticmethod
+    @pytest.mark.parametrize("bimaxwellian", [True, False])
+    def test_ordering_invariance(bimaxwellian):
+        r"""Test invariance to ordering of the bias and current values"""
+
+        sim = characteristic_simulated()
+
+        sim_result = swept_probe_analysis(
```

Comment:
This and `test_ion_mass_unit` seem to be the only ones that test `swept_probe_analysis` without expecting an error to be raised. These two tests depend on the same dataset `sim = characteristic_simulated()`. Do you think this is sufficiently exhaustive?
Testing over a range of plasma parameters, especially towards the limits of validity for these models, might be better. We could always merge and then open an issue on this.

Reply:
If I understand it correctly we're going to add actual experimental data to this a bit later, once we have the basic infrastructure going.

Reply:
Yeah, a simulated curve will inherently be tailored to the equations used in the analysis. We need the intricacies of real, experimental characteristics in the tests too.